components/ILIAS/LegalDocuments/classes/Administration.php

```diff
@@ -125,7 +133,46 @@ public function retrieveDocuments(): array
 
     public function retrieveIds(): array
     {
-        return $this->container->http()->wrapper()->post()->retrieve('ids', $this->container->refinery()->to()->listOf($this->container->refinery()->kindlyTo()->int()));
+        $ids = $this->container->http()->wrapper()->post()->retrieve(
```

Comment:
Please extract this into it's own (private) method as you are adding this code block at least 5 times.
E.g. Into something like this:
```php
private function retrieveDefaultOrSomeBetterNameIfYouSeeFit(string $key, $transformation, $default = null){
return $this->container->http()->wrapper()->post()->retrieve(
            $key,
            $this->container->refinery()->byTrying([
                $transformation,
                $this->container->refinery()->always($default)
            ])
        );
}
```

So you can just call:
```php
$this->retrieveDefaultOrSomeBetterNameIfYouSeeFit('ids', $this->container->refinery()->to()->listOf($this->container->refinery()->kindlyTo()->int()));
```

Or
```php
$list = $this->container->refinery()->to()->listOf(...);
$int = $this->container->refinery()->kindlyTo()->int();

$this->retrieveDefaultOrSomeBetterNameIfYouSeeFit('ids', $list($int));
...
$this->retrieveDefaultOrSomeBetterNameIfYouSeeFit('legal_document_id', $list($int), []);
```
Or something like that.